File: course/src/main/java/in/testpress/course/ui/CourseListActivity.java
package in.testpress.course.ui;

import android.content.Intent;
import android.os.Bundle;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;

import in.testpress.course.R;
import in.testpress.ui.BaseToolBarActivity;
import in.testpress.store.TestpressStore;
import android.app.Activity;


public class CourseListActivity extends BaseToolBarActivity {

    private CourseListFragment fragment;

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.testpress_container_layout);
        fragment = new CourseListFragment();
        fragment.setArguments(getIntent().getExtras());
        getSupportFragmentManager().beginTransaction().replace(R.id.fragment_container, fragment)
                .commitAllowingStateLoss();
    }

    @Override
    protected void onActivityResult(int requestCode, int resultCode, @Nullable Intent data) {
        super.onActivityResult(requestCode, resultCode, data);
        if (requestCode == TestpressStore.STORE_REQUEST_CODE && resultCode == Activity.RESULT_OK && data != null && !data.getBooleanExtra(TestpressStore.CONTINUE_PURCHASE, false)) {
            for (Fragment fragment : getSupportFragmentManager().getFragments()) {
                fragment.onActivityResult(requestCode, resultCode, data);
            }
        }
    }

    @Override
    public void onBackPressed() {
        if (fragment.onBackPress()){
            super.onBackPressed();
        }
    }
}

File: course/src/main/java/in/testpress/course/ui/CourseListActivity.kt
package `in`.testpress.course.ui

import android.app.Activity
import android.content.Intent
import android.os.Bundle
import `in`.testpress.course.R
import `in`.testpress.store.TestpressStore
import `in`.testpress.ui.BaseToolBarActivity

class CourseListActivity : BaseToolBarActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.testpress_container_layout)
        val fragment = CourseListFragment()
        fragment.arguments = intent.extras
        supportFragmentManager.beginTransaction().replace(R.id.fragment_container, fragment)
            .commitAllowingStateLoss()
    }

    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
        super.onActivityResult(requestCode, resultCode, data)
        if (requestCode == TestpressStore.STORE_REQUEST_CODE && resultCode == Activity.RESULT_OK){
            data?.let {  intentData ->
                if (!intentData.getBooleanExtra(TestpressStore.CONTINUE_PURCHASE, false)) {
                    supportFragmentManager.fragments.forEach { fragment ->
                        fragment.onActivityResult(requestCode, resultCode, intentData)
                    }
                }
            }
        }
    }
}